node/src/command.rs

```diff
@@ -186,20 +225,38 @@ pub fn run() -> Result<()> {
 				Ok(cmd.run(components.client, components.backend))
 			})
 		},
+		// TODO fix block import
 		Some(Subcommand::ExportGenesisState(params)) => {
 			let mut builder = sc_cli::LoggerBuilder::new("");
 			builder.with_profiling(sc_tracing::TracingReceiver::Log, "");
 			let _ = builder.init();
 
 			let chain_spec = cli.load_spec(&params.chain.clone().unwrap_or_default())?;
 			let state_version = Cli::native_runtime_version(&chain_spec).state_version();
-			let block: Block = generate_genesis_block(&chain_spec, state_version)
-				.map_err(|e| format!("{:?}", e))?;
-			let raw_header = block.header().encode();
-			let output_buf = if params.raw {
-				raw_header
-			} else {
-				format!("0x{:?}", HexDisplay::from(&block.header().encode())).into_bytes()
+
+			let output_buf = match chain_spec {
+				chain_spec if chain_spec.is_turing() => {
+					let block: turing_runtime::Block =
```

Comment:
separate functions for each of these possible? or would that make the ownership transfer terrible here?